### PR TITLE
Theorem 8.8

### PR DESCRIPTION
**Pull request by TashiWalde**
1) Introduce the notions of left fibration and inner fibration as right orthogonal maps wrt `{0} ⊂ Δ¹` and `Λ ⊂ Δ²`, respectively.

2) Show that the notions of naive left fibration and left fibration agree. In particular, every projection `∑ A, C -> A` of a covariant family `C : A -> U` is a left fibration (regardless of whether `A` is Segal).

3) Use the calculus of left orthogonal shapes to give a slick proof that every left fibration is an inner fibration.

4) Deduce that for every covariant family `C : A -> U` over a Segal type `A`, the total type `∑ C` is also Segal. This addresses #12 (dare I say closes?) unless @cesarbm03 (or anyone else) still wants to implement an alternative proof.

Notes:
- Relies on #88.
- Relies on `naiveextext`, since it relies on `is-right-orthogonal-to-shape-×`, which does; see #91.
- The proof of `is-naive-left-fibration-iff-is-covariant` could probably be refactored and shortened now, since the only hard part thereof should be a consequence of `extension-weakening`.

### CODE REVIEW

**Comment by jonweinb** [comment]
For a possible proof of 2.) that doesn't assume the base to be Segal see [[BW23, Proposition 6.1.1](https://higher-structures.math.cas.cz/api/files/issues/Vol7Iss1/BuchholtzWeinberger).

**Comment by TashiWalde** [comment]
> For a possible proof of 2.) that doesn't assume the base to be Segal see [[BW23, Proposition 6.1.1](https://higher-structures.math.cas.cz/api/files/issues/Vol7Iss1/BuchholtzWeinberger).
> 
> Edit: More precisely, the statement referred to is that any covariant family is inner.

The proof I implemented also does not use that the base is Segal.

**Review by jonweinb — APPROVED** [review]
That looks great!

**Comment by fizruk** [comment]
> Currently this does not properly compile for what seems to be a very weird reason: rzk complains that the variable naiveextext is declared as used but is actually unused. This is not true; and in fact rzk complains about the very same undeclared variable if I omit the uses (naiveextext). @fizruk, what am I missing here?

First, currently you get the following error:

```
unused variables
  naiveextext
declared as used in definition of
  is-segal-domain-inner-fibration-is-segal-codomain
```

This is because this definition does not actually rely on `naiveextext`, so `uses (naiveextext)` is not needed.
Once you remove that, everything typechecks on my machine with `rzk-0.6.6` (I did not push any fixes for that).

File: src/simplicial-hott/08-covariant.rzk.md [hunk 1]
```diff
@@ -220,6 +222,183 @@ As a sanity check we unpack the definition of `is-naive-left-fibration`.
...
+
+#end is-inner-fibration-is-left-fibration
+```
+
+Since the Segal types are precisely the local types with respect to `Λ ⊂ Δ¹`,
+we immediately deduce that in any left fibration `α : A' → A`,
+if `A` is a Segal type, then so is `A'`.
+
+```rzk title="Theorem 8.8, categorical version"
+#def is-segal-domain-inner-fibration-is-segal-codomain uses (naiveextext)
```
Comment (fizruk) [inline src/simplicial-hott/08-covariant.rzk.md:388]: The following change fixes typechecking:

```suggestion
#def is-segal-domain-inner-fibration-is-segal-codomain
```

Comment (TashiWalde) [reply src/simplicial-hott/08-covariant.rzk.md:388]: Thanks, I must have gotten confused and removed the wrong instance of `uses (naiveextext)`. This also made me realize that I mislabeled this term. Will fix this right away.

Comment (TashiWalde) [reply src/simplicial-hott/08-covariant.rzk.md]: Thanks, I must have gotten confused and removed the wrong instance of `uses (naiveextext)`. This also made me realize that I mislabeled this term. Will fix this right away.

**Comment by TashiWalde** [comment]
> > Currently this does not properly compile for what seems to be a very weird reason: rzk complains that the variable naiveextext is declared as used but is actually unused. This is not true; and in fact rzk complains about the very same undeclared variable if I omit the uses (naiveextext). @fizruk, what am I missing here?
> 
> First, currently you get the following error:
> 
> ```
> unused variables
>   naiveextext
> declared as used in definition of
>   is-segal-domain-inner-fibration-is-segal-codomain
> ```
> 
> This is because this definition does not actually rely on `naiveextext`, so `uses (naiveextext)` is not needed. Once you remove that, everything typechecks on my machine with `rzk-0.6.6` (I did not push any fixes for that).

Thanks! I must have gotten confused and removed the wrong instance of `uses (naiveextext)`.

**Review by fizruk — DISMISSED** [review]

**Review by emilyriehl — APPROVED** [review]

**Pull request merged**